pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/typedefinition/JavaTypeDefinitionSimple.java

```diff
@@ -243,7 +245,21 @@ public boolean isArrayType() {
     public String toString() {
         return new StringBuilder("JavaTypeDefinition [clazz=").append(clazz)
                 .append(", definitionType=").append(getDefinitionType())
-                .append(", genericArgs=").append(genericArgs)
+                .append(", genericArgs=").append(genericArgs.stream().map(new Function<JavaTypeDefinition, String>() {
```

Comment:
stream() is a java8 feature, isn't it? We are still on java7... I'm wondering, why the ci build doesn't fail...

Reply:
It seems, we run into the "wrong bootclasspath" problem - it compiles fine as java7 but doesn't run later with java7 (e.g. class not found: java.util.function.Function).

For releasing 6.0.0 I switched back from java9 to java8, so that pmd-scala is included. When switching to java9 we removed the maven toolchain support (which would avoid the wrong bootclasspath problem) and used the new java9 "-release" feature.

I'll look into a solution like [this](https://stackoverflow.com/questions/9164625/maven-use-jdk-7-to-compile-for-jvm-5) tomorrow.

Reply:
Interesting! good we find this! I'll rework the PR ASAP, but glad we can tackle it to close the door for future mistakes

Reply:
See #842